rust-connectors/sources/http/src/bin/main.rs

```diff
@@ -89,11 +69,30 @@ async fn main() -> Result<()> {
         if let Some(ref body) = opts.body {
             req = req.body(body.clone());
         }
-        let response = req.send().await?;
-        let response_text = response.text().await?;
+        let response = req.send().await.map_err(|e| Error::Request(e))?;
+
+        let response_version = format!("{:?}", response.version());
+        let response_status = response.status().to_string();
+        let response_headers_full = ::http::formatter::format_reqwest_headers(response.headers());
+        let response_headers_count = response.headers().len();
+
+        let response_body = response.text().await.map_err(|e| Error::ResponseBody(e))?;
+
+        let record_out = match opts.output_format.as_str() {
+            "full" => ::http::formatter::format_full_record(
+                &response_version,
+                &response_status,
+                response_headers_count,
+                &response_headers_full,
+                &response_body,
+            ),
+
+            _ => response_body,
```

Comment:
I think that we should fail if an invalid format is used

Reply:
It's not non-exhaustive so generally if doing default StructOpt on strict I typically prefer this pattern to declare and align the default

If it would be non-exhaustive e.g. with public interface that exposes it outside then I would agree but it's implementation detail so this pattern imo would work better w/ non-exhaustive

happy to adopt any pattern however

Reply:
I made a bad mistake - output_format was non-exhaustive not sure what I was thinking. I added a panic! into _ - 

Was there anything proper I can signal connector init error?